pkg/trace/agent/agent.go

```diff
@@ -190,6 +190,37 @@ func (a *Agent) loop() {
 	}
 }
 
+// sanitizeSpan applies some additional sanitization, including obfuscation,
+// truncation, and adding metadata based on the configured tags.
+func (a *Agent) sanitizeSpan(span *pb.Span, chunk *pb.TraceChunk, clientComputedTopLevel bool) {
+	for k, v := range a.conf.GlobalTags {
+		if k == tagOrigin {
+			chunk.Origin = v
+		} else {
+			traceutil.SetMeta(span, k, v)
+		}
+	}
+	if a.ModifySpan != nil {
+		a.ModifySpan(span)
+	}
+	a.obfuscateSpan(span)
+	Truncate(span)
+	if clientComputedTopLevel {
+		traceutil.UpdateTracerTopLevel(span)
+	}
+}
+
+// setRootSpanTags sets up any necessary tags on the root span.
+func (a *Agent) setRootSpanTags(root *pb.Span) {
```

Comment:
> `setRootSpanTags`

Is that actually what it does? It looks like it sets sample rates?

Reply:
Setting the rates means to set certain tags (metrics).

The comment is derived from the original code:
```
{
    // this section sets up any necessary tags on the root:
    ...
}
```